sources/cadvisor.go

```diff
@@ -46,8 +49,12 @@ func (self *cadvisorSource) getAllCadvisorData(hostname, ip, port, container str
 	if err != nil {
 		return
 	}
+	numStats := int(self.pollDuration / time.Second)
+	if time.Since(self.lastQuery) > self.pollDuration {
+		numStats = int(time.Since(self.lastQuery) / time.Second)
+	}
 	allContainers, err := client.SubcontainersInfo("/",
-		&cadvisor.ContainerInfoRequest{NumStats: int(self.pollDuration / time.Second)})
+		&cadvisor.ContainerInfoRequest{NumStats: numStats})
```

Comment:
This assumes that each stat is collected at 1s interval which is not true. This is something broken in the cAdvisor API that I've meant to fix. Will probably be fixed for API v2.0.

Does not affect this PR :)